### ISSUE TITLE
Feature request: built-in icon stacking

### ISSUE THREAD

**Issue by adrfantini** [duskmoon314/typst-fontawesome#12]
Icon stacking is a core feature of FontAwesome. It is documented and encouraged: https://docs.fontawesome.com/web/style/stack

this extension could provide its own icon stacking functionality. There is no official way to do this in Typst, but here is an example:
```typst
#let z-stack(..items) = { // Source: https://github.com/typst/typst/issues/682#issuecomment-2041124521
  grid(
    columns: items.pos().len() * (1fr,),
    column-gutter: -100%,
    rows: 1,
    ..items
  )
}
```

This works well in my tests, for example stacking `integral` and a red `slash`:
```typst
#import "@preview/fontawesome:0.2.1": *
#fa-icon = fa-icon.with(font: "Font Awesome 6 Pro", hyphenate: false)

#z-stack(
  fa-icon("integral"),
  fa-icon("slash", fill: red)
)
```
![image](https://github.com/user-attachments/assets/b7ff0899-db32-41ec-9e6b-48f11696d54d)

**Comment by duskmoon314** [comment]
Interesting! I have yet to think about this.

Maybe the API would be like this (haven't tested whether this is possible):

```typst
fa-stack(
  ("integral", ("slash", (fill: red))), // list of icons and their args
  font: "Font Awesome 6 Pro"            // args that passed to fa-icon
)
```

Font Awesome provides some features through CSS. Is there any feature that should be taken care of in typst?

Also, feel free to open a PR if you've implemented something.

**Comment by duskmoon314** [comment]
@adrfantini I have implemented an `fa-stack` function in the PR #14. Could you review it and share your thoughts?
